qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteSnippetCompletionInIfSectionTest.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute snippet completion in #if section.
+ *
+ */
+public class QuteSnippetCompletionInIfSectionTest {
+
+	@Test
+	public void singleIf() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if", //
+						"{#if ${1:condition}}" + System.lineSeparator() + //
+								"	$0" + System.lineSeparator() + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void singleElse() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if-else", //
+						"{#if ${1:condition}}" + System.lineSeparator() + //
+								"	$2" + System.lineSeparator() + //
+								"{#else}" + System.lineSeparator() + //
+								"	$0" + System.lineSeparator() + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void singleElseIf() throws Exception {
+		String template = "|";
+		testCompletionFor(template, //
+				true, // snippet support
+				c("if-elseif", //
+						"{#if ${1:condition}}" + System.lineSeparator() + //
+								"	$2" + System.lineSeparator() + //
+								"{#else if ${3:condition}}" + System.lineSeparator() + //
+								"	$4" + System.lineSeparator() + //
+								"{#else}" + System.lineSeparator() + //
+								"	$0" + System.lineSeparator() + //
+								"{/if}", //
+						r(0, 0, 0, 0)));
+	}
+
+	@Test
+	public void nestedElse() throws Exception {
+		String template = "{#if condition}" + System.lineSeparator() + //
+				"|" + System.lineSeparator() + //
+				"{/if}";
+		testCompletionFor(template, //
+				true, // snippet support

```

Comment:
It is important to check that no future nested section will appear in the completion.
To do that you must check the completion items size is the number of snippets + your 2 nested snippets.

See à sample at https://github.com/redhat-developer/quarkus-ls/blob/b5a04317e0363a33b08303d1635ef1a31eb54f30/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInSectionTagTest.java#L123

I suggest you remove your last test and move the check of elseif in this test

Reply:
In other words:

```java
@Test
public void nestedIf() throws Exception {
String template = "{#if condition}\n" + //
		"|\n"  + //
		"{/if}";
testCompletionFor(template, //
		true, // snippet support,
		SECTION_SNIPPET_SIZE /* #each, #if, ... */+ 2 /* #else, #else if*/,
		c("else", "{#else}$0", r(1, 0, 1, 0)), //
		c("elseif", "{#else if ${1:condition}}$0", r(1, 0, 1, 0)));
}
```

And this test is nice because it higlight a bug here https://github.com/redhat-developer/quarkus-ls/pull/656/files#diff-f8df902c2cf37805c3a21be3609ea2b71e36e8f4db6db3f6f5f3f41594e50dd4R53 indeed the node where completion was triggered is a Text and not a Section, you should support the both. And write an another test like 

```java
@Test
public void nestedIf() throws Exception {
String template = "{#if condition}\n" + //
		"{#el|\n"  + //
		"{/if}";
testCompletionFor(template, //
		true, // snippet support,
		SECTION_SNIPPET_SIZE /* #each, #if, ... */+ 2 /* #else, #else if*/,
		c("else", "{#else}$0", r(1, 0, 1, 0)), //
		c("elseif", "{#else if ${1:condition}}$0", r(1, 0, 1, 0)));
}
```